### ISSUE TITLE
Solr 8.6.3 Release

### ISSUE THREAD

**Issue by hossman** [docker-solr/docker-solr#349]
Solr 8.6.3 has not yet hit all the mirrors, but in preparation here is a patch generated using `tools/update.sh`.

I have verified that Jason's GPG key is valid, that `tools/build_all.sh` works, and that the following all pass...

* `./test.sh 8.6`
* `./test.sh 8.6.3`
* `./test.sh 8.6.3-slim`

[docker-solr-8.6.3.patch.txt](https://github.com/docker-solr/docker-solr/files/5344876/docker-solr-8.6.3.patch.txt)

**Comment by dsmiley** [comment]
Good to see you here Hoss!  LOL I've never seen a patch file attached to a GitHub issue before -- that's what PRs are for :-)  I know you have a principled objection to ASF projects relying on GitHub infrastructure but... you are already here as you have to be because docker-solr fundamentally lives here so... can you bring yourself to file a PR?

**Comment by dsmiley** [comment]
I misjudged your rationale; I now realized from another message you sent that you are unfamiliar with how to do PRs.  Basically, fork and push to your fork, then file a PR.  I promise it's really quite easy... albeit unfamiliar at first.  "Fork" seems heavy but it's not at all.

**Comment by hossman** [comment]
it's less about principle and more about the stupidity of GH insisting that there's no way to contribute to a project other then to create my own personal private (GH hosted) clone of that project, set my new GH hosted clone as a remote on my local co, push changes to that remote clone, and *THEN* request that those changes be made to the official repo ... then way, and only after the changes are made i can go delete that GH hosted clone i don't care about and don't wnat to have sitting out there forever getting stale.

but sure, whatever PR created.

**Comment by dsmiley** [comment]
RE "getting stale", I suggest thinking of GitHub forks as purely there to satisfy PRs, and rarely other purposes.  I could care less how "stale" my forks are.  I will never maintain them; they are only needed as a place to stage a PR to an upstream repo.  I do this even for upstreams that I have access rights to because it avoids polluting the branch namespace of the upstream repo -- clearly a problem of lucene-solr.  GitHub foots the hosting bill, and no doubt they are clever in re-using internal storage with the project you are forking so the overhead is minimal.  I sympathize that the whole result (Git & GitHub) on the PR workflow has one to two too-many steps relative to a simpler ideal that could be much easier if history played out differently, or if GitHub had tried to solve somehow.... but it is what it is.  At least once you create the fork and add the remote, you needn't do that again!

Thanks for the PR.  The PR is also tied into a CI system that allowed more verification that the PR was good.
The final step is another PR to Docker Inc:  https://github.com/docker-library/official-images/pull/8663  which will take a day, maybe two.

**Comment by chatman** [comment]
I think we should write a bot for those who hate forks (like me). The bot
could create a branch from a patch and create a PR. It irritates my
reviewers if I don't raise a PR (Noble has called me out on several
occasions), but it irritates me to create a upstream branch or update my
fork for a simple fix. WDYT, Hoss?

**Comment by dsmiley** [comment]
Why would the input be a "patch"?  Wouldn't it be a branch?  Would creating of the PR auto-create a fork if you don't have one?

What I wish for is a single-action thing in my IDE to take my current branch (or some other configured branch or some changelist currently) and file the PR for me.  If I choose the changelist method, it needs to commit to a new feature branch.  It'd need to ask me for things like a commit message etc. and possibly branch name but there are too many steps today.  I live with it.

**Comment by chatman** [comment]
Workflow I had in mind was that contributor attaches a patch file to a
GitHub issue or jira issue on Solr, a bot created a branch on its fork (bot
has his own fork) and raises the PR. This can completely replace the yetus
thing we have, I guess. And the patch people can be happy as much as PR
lovers.

**Comment by hossman** [comment]
For other folks who may be following along who may be equally unfamiliar with the whole process as i am...

I was confused as to where things stand after my PR got merged; but in tracing through the full steps to get the "official" repo updated, i found these commits & PRs from smiley are in progress and waiting on the approval into docker-library/official-images ...

* https://github.com/docker-solr/official-images/commit/bbce5d5a685cd8701a4a677fae5369845b70ea57
* https://github.com/docker-library/official-images/pull/8870
